### ISSUE TITLE
State of the tokio-tls crate

### ISSUE THREAD

**Issue by dmolokanov** [tokio-rs/tls#7]
Could you please clarify what is the state of [tokio-tls](https://github.com/tokio-rs/tokio/tree/master/tokio-tls) and difference from the current crate?

I was looking for a way to get a peer certificate for my TLS server and haven't found a way to do so with the `tokio-tls` which I'm using right now. But in the current crate, you merged a PR to support this in PR #6. But this crate wasn't released yet.

Now, I'm confused about which version I should use going forward. Do you have any plans to backport changes from #6 to the `tokio-tls`? Or as alternative to release `tokio-native-tls` instead? Any estimates on release date if yes?

Really appreciate any information. Thanks!

**Comment by dmolokanov** [comment]
As we discussed in Discord the main stopper for tokio team to make all tests passed for macos. While all tests are passing on the Mojave they area failing on Catalina.

I did some digging into cert is used for smoke tests and it has following validity period
```Validity
            Not Before: Aug 14 17:00:03 2016 GMT
            Not After : Aug 12 17:00:03 2026 GMT
``` 
And that is what Apple doesn't consider it secure enough because it is valid for more than 875 days. As pointed in the [issue](https://github.com/clux/kube-rs/issues/89#issuecomment-568269142)
